snowshu/adapters/target_adapters/snowflake_adapter/utils.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+import pendulum
+import snowflake.connector
+
+
+logger = logging.getLogger(__name__)
+
+
+def connect_to_database(credentials):
+    try:
+        conn = snowflake.connector.connect(
+            user=credentials["user"],
+            password=credentials["password"],
+            account=credentials["account"],
+            database=credentials["database"],
+            role=credentials["role"],
+        )
+        return conn
+    except snowflake.connector.errors.Error as e:
+        logger.info(f"Failed to connect to Snowflake: {e}")
+        return None
+
+
+def rename_database(cursor: snowflake.connector.Cursor, old_name: str, new_name: str):
+    """Rename a database, dropping the new name if it already exists."""
+    try:
+        cursor.execute(f"ALTER DATABASE {old_name} RENAME TO {new_name}")
```

Comment:
can we try to execute both statements in one transaction?
drop if exists + actual rename

Reply:
potentially we could, but I think current approach is more explicit and easier to understand and debug in case of anything, if I give both statements I won't get any return if we actually dropped something cause - we could dig into cursor return object, but it will take time now and I don't really see any particular profit from it, what do you think? maybe I'm missing something @ArtemHU

Reply:
the end result will remain consistent in both scenarios, even though the log provides sufficient information to comprehend what was dropped.